firstvoices/backend/tests/test_async/test_async.py

```diff
@@ -0,0 +1,129 @@
+        user = factories.get_app_admin(role=AppRole.SUPERADMIN)
+        self.client.force_authenticate(user=user)
+
+        response = self.client.post(
+            f"{self.get_detail_endpoint('missing-slug')}dictionary-cleanup/preview"
+        )
+        assert response.status_code == 404
+
+    @pytest.mark.django_db
+    def test_recalculate_preview_get_403(self, site):
+        user = factories.get_non_member_user()
+        self.client.force_authenticate(user=user)
+
+        response = self.client.get(
+            f"{self.get_detail_endpoint(site.slug)}dictionary-cleanup/preview"
+        )
+        assert response.status_code == 403
+
+    @pytest.mark.django_db
+    def test_recalculate_preview_post_403(self, site, alphabet):
+        user = factories.get_non_member_user()
+        self.client.force_authenticate(user=user)
+
+        response = self.client.post(
+            f"{self.get_detail_endpoint(site.slug)}dictionary-cleanup/preview"
+        )
+        assert response.status_code == 403
+
+    @pytest.mark.django_db
+    def test_recalculate_preview_get_empty(self, site, alphabet):
+        user = factories.get_app_admin(role=AppRole.SUPERADMIN)
+        self.client.force_authenticate(user=user)
+        assert user.has_perm(self.SUPERADMIN_PERMISSION, site)
+
+        response = self.client.get(
+            f"{self.get_detail_endpoint(site.slug)}dictionary-cleanup/preview"
+        )
+        response_data = json.loads(response.content)
+        assert response.status_code == 404
+        assert response_data == {
+            "message": "No recalculation results found for this site."
+        }
+
+    @pytest.mark.django_db(transaction=True, serialized_rollback=True)
+    def test_recalculate_preview_post(
+        self, celery_worker, site, alphabet, django_db_serialized_rollback
+    ):
+        user = factories.get_app_admin(role=AppRole.SUPERADMIN)
+        self.client.force_authenticate(user=user)
+        assert user.has_perm(self.SUPERADMIN_PERMISSION, site)
+
+        factories.DictionaryEntryFactory.create(site=site, title="test")
+
+        response = self.client.post(
+            f"{self.get_detail_endpoint(site.slug)}dictionary-cleanup/preview"
+        )
+
+        response_data = json.loads(response.content)
+        assert response.status_code == 201
+        assert response_data == {"message": "Successfully saved recalculation results"}
+
+    @pytest.mark.django_db(transaction=True, serialized_rollback=True)
+    def test_recalculate_preview_full(
+        self, celery_worker, site, alphabet, django_db_serialized_rollback
+    ):
+        user = factories.get_app_admin(role=AppRole.SUPERADMIN)
+        self.client.force_authenticate(user=user)
+        assert user.has_perm(self.SUPERADMIN_PERMISSION, site)
+
+        factories.DictionaryEntryFactory.create(site=site, title="test")
+
+        response_post = self.client.post(
+            f"{self.get_detail_endpoint(site.slug)}dictionary-cleanup/preview"
```

Comment:
Seems like it would be worth overriding get_detail_endpoint to return this whole url